include/shark/Algorithms/DirectSearch/Operators/Hypervolume/HypervolumeContribution2D.h

```diff
@@ -0,0 +1,173 @@
+		*heapEnd = std::make_pair((front[2].f1 - front[1].f1) * (front[0].f2 - front[1].f2),front[1].index);
+		++heapEnd;
+		for(std::size_t i = 2; i < front.size()-1;++i){
+			double contribution = (front[i+1].f1 - front[i].f1)*(front[i-1].f2 - front[i].f2);
+			*heapEnd = std::make_pair(contribution,front[i].index);
+			++heapEnd;
+			std::push_heap(heapStart,heapEnd,pointComp);
+			
+			if(heapEnd == bestK.end()){
+				std::pop_heap(heapStart,heapEnd,pointComp);
+				--heapEnd;
+			}
+		}
+		bestK.pop_back();//remove the k+1th element
+		std::sort_heap(heapStart,heapEnd,pointComp);
+		
+		std::vector<std::size_t> index;
+		for(auto const& entry: bestK){
+			index.push_back(entry.second);
+		}
+		return index;
+	}
+public:
+	/// \brief Returns the index of the points with smallest contribution.
+	///
+	/// \param [in] points The set \f$S\f$ of points from which to select the least contributor.
+	/// \param [in] k The number of points to select.
+	/// \param [in] referencePoint The reference point \f$\vec{r} \in \mathbb{R}^2\f$ for the hypervolume calculation, needs to fulfill: \f$ \forall s \in S: s \preceq \vec{r}\f$.
+	template<typename Set, typename VectorType>
+	std::vector<std::size_t> least( Set const& points, std::size_t k, VectorType const& referencePoint)const{
+		std::vector<Point > front;
+		front.emplace_back(0,referencePoint[1],points.size()+1);//add reference point
+		for(std::size_t i  = 0; i != points.size(); ++i)
+			front.emplace_back(points[i][0],points[i][1],i);
+		front.emplace_back(referencePoint[0],0,points.size()+1);//add reference point
+		std::sort(front.begin()+1,front.end()-1);
+		
+		return bestContributors(front,k,[](double con1, double con2){return con1 < con2;});
+	}
+	
+	/// \brief Returns the index of the points with smallest contribution.
+	///
+	/// As no reference point is given the edge points can not be computed and are not selected.
+	///
+	/// \param [in] points The set \f$S\f$ of points from which to select the least contributor.
+	/// \param [in] k The number of points to select.
+	template<typename Set>
+	std::vector<std::size_t> least( Set const& points, std::size_t k)const{
+		std::vector<Point > front;
+		for(std::size_t i  = 0; i != points.size(); ++i)
+			front.emplace_back(points[i][0],points[i][1],i);
+		std::sort(front.begin()+1,front.end()-1);
+		
+		return bestContributors(front,k,[](double con1, double con2){return con1 < con2;});
+	}
+	
+	/// \brief Returns the index of the points with largest contribution.
+	///
+	/// \param [in] points The set \f$S\f$ of points from which to select the least contributor.
+	/// \param [in] referencePoint The reference point \f$\vec{r} \in \mathbb{R}^2\f$ for the hypervolume calculation, needs to fulfill: \f$ \forall s \in S: s \preceq \vec{r}\f$.
+	template<typename Set, typename VectorType>
+	std::vector<std::size_t> largest( Set const& points, std::size_t k, VectorType const& referencePoint)const{
```

Comment:
"largest" is the opposite of "smallest", "most" is the opposite of "least". Another possible pair would be worst/best, which is probably closest to what we want here. So I propose to rename least -> worst and largest -> best.